bundles/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/device/PowerStripDevice.java

```diff
@@ -73,7 +68,18 @@ public void refreshedDeviceState(@Nullable DeviceState deviceState) {
     }
 
     @Override
-    public State updateChannel(ChannelUID channelUid, DeviceState deviceState) {
+    protected State getOnOffState(final DeviceState deviceState) {
+        // Global On/Off state is determined by the combined state of all sockets. If 1 socket is on, the state is on.
+        for (int i = 0; i < childIds.size(); i++) {
```

Comment:
```suggestion
        return deviceState.getSysinfo().getChildren().stream().parallel().filter(e -> OnOffType.ON.equals(e.getState()).findAny().isPresent();
```

Reply:
Ah, yes that can be done using streams. Added it with a slightly different variant as it requires a  State.

Reply:
Nice